test-support/page-object/custom-helper.js

```diff
@@ -0,0 +1,30 @@
+import Ember from 'ember';
+import { build } from './build';
+import { qualifySelector } from './helpers';
+
+export function customHelper(userDefinedFunction) {
+  return function(selector, options) {
+    // def
+    return {
+      buildPageObjectAttribute: function(key, parent) {
+        // def
+        return function() {
+          // eval
+          let scopedSelector = qualifySelector(parent.scope, selector);
+
+          let customHelperRes = userDefinedFunction(scopedSelector, options);
+
+          if ($.isPlainObject(customHelperRes)) {
+            return build($.extend(customHelperRes, { scope: scopedSelector }),
+                         key,
+                         parent);
+          } else if ($.isFunction(customHelperRes)) {
+            return customHelperRes(scopedSelector);
```

Comment:
Following the spec we agreed on, we must pass the result of calling `qualifySelector(parent.scope, selector)` to get the parent's scope when the user defined function is called.

```
var input = customHelper(function(selector, options) {
  let hasError = $(selector).parent().hasClass('has-error');

  return {
    value: value(selector),
    hasError: function() {
      return hasError;
    }
  };
});

var page = PageObject.build({
  scope: 'form',
  userName: input('#userName')
});
```
In this case, the input function should receive `form #userName` as a parameter.